pkg/virt-handler/domain.go

```diff
@@ -36,11 +38,10 @@ type DomainDispatch struct {
 }
 
 func (d *DomainDispatch) Execute(indexer cache.Store, queue workqueue.RateLimitingInterface, key interface{}) {
-	obj, exists, err := indexer.GetByKey(key.(string))
-	if err != nil {
-		queue.AddRateLimited(key)
-		return
-	}
+	obj, exists, _ := indexer.GetByKey(key.(string))
+	// GetByKey can never return anything but nil for the error
```

Comment:
Why? There is an interface which says, return an error if there is one, but that does not mean that every implementation necessarily is written in a way that there can happen an error to report.

The most extreme examples: Think of an in memory map, which can't have an error when looking up a key. It is a valid implementation of a store and when accessed through the interface, you always return nil as an error. Now think of a file based store. When looking something there up, it can fail.

The interface is the contract I would want to implement. I don't think it is nice, to assume a concrete implementation behind an interface.